indico/modules/events/papers/models/revisions.py

```diff
@@ -185,10 +185,15 @@ def get_reviews(self, group=None, user=None):
 
     def get_reviewed_for_groups(self, user, include_reviewed=False):
         from indico.modules.events.papers.models.reviews import PaperTypeProxy
-        reviewed_for = {x.type for x in self.reviews if x.user == user} if include_reviewed else set()
-        if self.paper.cfp.content_reviewing_enabled and user in self.paper.cfp.content_reviewers:
+        from indico.modules.events.papers.util import is_type_reviewing_possible
+
+        cfp = self.paper.cfp
+        reviewed_for = set()
+        if include_reviewed:
+            reviewed_for = {x.type for x in self.reviews if x.user == user and is_type_reviewing_possible(cfp, x.type)}
```

Comment:
Not sure how exactly this function i used, but if it shows the reviewing types in which the user has already reviewed checking if they are still within the deadline seems incorrect here.

Reply:
This is the list of available groups (content, layout). So user will be available to review in any of the groups that are in this set. I don't want to include any of these even if there is an existing review.

Reply:
the name and iteration over `self.reviews` made me assume it's also used to show the groups in which you *already* reviewed

Reply:
As far as I can tell there is another place where we return the list of groups that had been reviewed.

```python
    def get_reviewer_render_data(self, user):
        groups = self.get_reviewed_for_groups(user, include_reviewed=True)
        reviews = {x.group: x for x in self.get_reviews(user=user)}
        reviewed_groups = {x.group for x in reviews.itervalues()}
        missing_groups = groups - reviewed_groups
        return {'groups': groups,
                'missing_groups': missing_groups,
                'reviewed_groups': reviewed_groups,
                'reviews': reviews}
```

Reply:
OK, I guess you tested that it works correctly anyway ;)

Reply:
Hah, of course as always :D I hope I didn't miss anything.